model-optimizer/mo/front/extractor.py

```diff
@@ -766,46 +783,38 @@ def extract_node_attrs(graph: Graph, extractor: callable):
     return graph
 
 
-def extract_port_from_string(node_name: str):
-    """
-    Extracts port and node name from string
-
-    Raises if node name was not provided in the expected format:
-    NODE:OUT_PORT
-        or
-    IN_PORT:NODE
-
-    :param node_name: string value provided by user
-    :return: node name, input port and output port
-    """
-    parts = node_name.split(':')
-    if len(parts) > 2:
-        raise Error("Please provide only one port number for {}. Expected format is NODE:OUT_PORT or IN_PORT:NODE, "
-                    "where IN_PORT and OUTPUT_PORT are integers".format(node_name))
-    if len(parts) == 1:
-        return node_name, None, None
-    else:
-        in_port, out_port, name = None, None, None
-        try:
-            in_port, name = int(parts[0]), parts[1]
-        except ValueError:
-            try:
-                out_port, name = int(parts[1]), parts[0]
-            except ValueError:
-                raise Error("Non integer port number in {}. Expected format is NODE:OUT_PORT or IN_PORT:NODE, where "
-                            "IN_PORT and OUTPUT_PORT are integers".format(node_name))
-        return name, in_port, out_port
-
-
-def get_node_id_with_ports(graph: Graph, name: str):
+def get_node_id_with_ports(graph: Graph, node_name: str):
     """
     Extracts port and node ID out of user provided name
     :param graph: graph to operate on
-    :param name: user provided node name
+    :param node_name: user provided node name
     :return: node ID, direction of port ('in', 'out', 'port') and port number or None
     """
-    node_name, in_port, out_port = extract_port_from_string(name)
-    node_id = graph.get_node_id_by_name(node_name)
+    node_names = [n.soft_get('name', n.id) for n in graph.get_op_nodes()]
+    found_names = []
+    for name in node_names:
+        regexp = r'(\d*:)?(' + name + r')(:\d*)?'
+        match = re.search(regexp, node_name)
+        if match and match.group() == node_name:
+            in_port = None
+            out_port = None
+            if match.group(1) and match.group(3):
+                log.warning('Skipping the case with both in and out port specified, only one port can be specified')
+                continue
+            if match.group(1):
+                in_port = int(match.group(1).replace(':', ''))
+            if match.group(3):
+                out_port = int(match.group(3).replace(':', ''))
+            found_names.append((in_port, out_port, name))
+    if len(found_names) == 0:
+        raise Error('No node with name {}'.format(node_name))
+    if len(found_names) > 1:
+        raise Error('Name collision was found, there are several nodes for mask "{}": {}. '
```

Comment:
The message is not clear enough. I would like to see some examples.

Reply:
Created story CVS-35723 for that problem.